### PR TITLE
PAYARA-988 Add asadmin command to configure the hogging threads heathcheck service

### PR DESCRIPTION
**Pull request by smillidge**

### CODE REVIEW

**Comment by smillidge** [comment]
jenkins test please

File: nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HoggingThreadsConfigurer.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    protected Logger logger;
+
+    @Inject
+    HealthCheckService healthCheckService;
+
+    @Param(name = "enabled", optional = false)
+    private Boolean enabled;
+
+    @Param(name = "time", optional = true)
+    @Min(value = 1, message = "Time period must be 1 or more")
```
Comment (Pandrex247) [inline nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HoggingThreadsConfigurer.java]: "Time period must be greater than 1" ;)

File: nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HoggingThreadsConfigurer.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+    @Param(name = "target", optional = true, defaultValue = "server-config")
+    protected String target;
+
+    @Inject
+    ServerEnvironment server;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+
```
Comment (Pandrex247) [inline nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HoggingThreadsConfigurer.java]: Random whitespace

File: nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HoggingThreadsConfigurer.java (hunk 1 above)
Comment (smillidge) [inline nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HoggingThreadsConfigurer.java]: must be greater than or equal to 1 ;)

Comment (Pandrex247) [inline nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HoggingThreadsConfigurer.java]: Damn! Out-grammared!

File: nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HoggingThreadsConfigurer.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+            if (dynamic) {
+                if (server.isDas()) {
+                    if (targetUtil.getConfig(target).isDas()) {
+                        HoggingThreadsChecker checkerByType = healthCheckServiceConfiguration.getCheckerByType(HoggingThreadsChecker.class);
+                        service.setOptions(service.constructOptions(checkerByType));
+                        healthCheckService.registerCheck(checkerByType.getName(), service);
+                        healthCheckService.reboot();
+                    }
+                } else {
+                    // it implicitly targetted to us as we are not the DAS
```
Comment (Pandrex247) [inline nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HoggingThreadsConfigurer.java]: Typo: targeted

Comment (Pandrex247) [inline nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/HoggingThreadsConfigurer.java]: Typo? It's

**Pull request merged**